### PR TITLE
Fix setting extended header when importing MRC

### PR DESCRIPTION
**Pull request by uermel**
Relates to
#166

## Description
The property `MrcObject.extended_header` may return `None`, or values of type `np.recarray` or `np.ndarray`. In contrast to python lists, `np.ndarray` is not truthy, so comparison with any `np.ndarray` will raise exceptions. In order to circumvent, we ensure that the result of `MRCReader.get_mrc_extended_header()` is a np.array and then convert to a python list that enables truthy/falsy comparison. 

Covers the following cases: 
1. do set extended header on filled `np.ndarray` > `np.array(val: np.ndarray).tolist() -> list`
2. do set extended header on filled `np.recarray` > `np.array(val: np.recarray).tolist() -> list`
3. skip setting extended header on empty `np.ndarray` > `np.array(np.empty(0)).tolist() == []`
4. skip setting extended header on `None` > `np.array(None).tolist() == None`

### CODE REVIEW

File: ingestion_tools/scripts/common/image.py [hunk 1]
```diff
@@ -353,7 +353,13 @@ def update_headers(self, mrcfile: MrcFile, header_mapper, voxel_spacing):
...
                 header.nsymbt = old_header.nsymbt
                 header.exttyp = old_header.exttyp
-                if ext := self.volume_reader.get_mrc_extended_header():
+                ext = self.volume_reader.get_mrc_extended_header()
+                # In order to cover the following cases:
+                # 1. do set on filled np.ndarray > np.array(val: np.ndarray).tolist() -> list
+                # 2. do set on filled np.recarray > np.array(val: np.recarray).tolist() -> list
+                # 3. skip setting on empty np.ndarray > np.array(np.empty(0)).tolist() == []
+                # 4. skip setting on None > np.array(None).tolist() == None
+                if np.array(ext).tolist():
```
Comment (manasaV3) [inline ingestion_tools/scripts/common/image.py:362]: Should the headers `exttyp` also be updated accordingly?

Comment (uermel) [reply ingestion_tools/scripts/common/image.py:362]: It is being updated in line 355, no?

Comment (manasaV3) [reply ingestion_tools/scripts/common/image.py:362]: Yes, but if we are not setting the extended header in some cases where there is an extended header in the original mrc file. Should that also reflect in the `exttyp` which is currently set to the value as original mrc file.

Comment (uermel) [reply ingestion_tools/scripts/common/image.py:362]: I see, so moving that inside of the if-clause would solve this issure, correct?

Comment (manasaV3) [reply ingestion_tools/scripts/common/image.py:362]: Yeah, I believe so. One quick sanity check, what would be the expected value for that header when there is no extended_header?

Comment (uermel) [reply ingestion_tools/scripts/common/image.py:362]: `header.exttyp = b'MRCO'`
`header.nsymbt = 0`

Comment (manasaV3) [reply ingestion_tools/scripts/common/image.py:362]: Sorry, in that case, should we reset only the `nsymbt` inside the if condition?

Comment (uermel) [reply ingestion_tools/scripts/common/image.py:362]: yep, I thought this has a default set somewhere else. I'll make the changes as requested.

Comment (uermel) [reply ingestion_tools/scripts/common/image.py:362]: On reflection, I think it's not necessary to set the extended header type, but we can set the extended header size to zero explicitely if there is no extended header.

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**